firstvoices/backend/search/query_builder.py

```diff
@@ -32,10 +41,14 @@ def get_cleaned_search_term(q):
     return q.strip()
 
 
-def get_search_query(q=""):
+def get_search_query(q="", site_slug=""):
     search_query = get_search_object()
     search_term = get_cleaned_search_term(q)
 
+    if site_slug:
+        site_slug = validate_site(site_slug)
+        search_query = search_query.filter("term", site_slug=site_slug)
+
```

Comment:
This seems like a good time to split out the different parts of the query into separate functions. Like get_site_query and get_term_query could return Q objects, and the job of get_search_query could be to assemble them based on the various conditions and inputs

Reply:
I've refactored the main function into smaller functions, but the logic to assemble Q objects still rest inside the specific functions. The parent function checks if the param is present, if yes, it passes the param to the required function and expects a modified search query in return with added filter/query.